### PR TITLE
Add how to start near node from backup

### PR DESCRIPTION
**Pull request by mhalambek**

### CODE REVIEW

**Review by chefsale — CHANGES_REQUESTED** [review]
While here can you please also fix the Running Official Node on GCP part in the same file. The requirements are wrong.

**Review by chefsale — APPROVED** [review]
Looks good now, one last nit: SSD instead of persistent storage

File: docs/develop/node/running-a-node.md [hunk 1]
```diff
@@ -188,3 +188,54 @@ To check logs call: `nearup logs` or `nearup logs --follow`
 >Got a question?
 <a href="https://stackoverflow.com/questions/tagged/nearprotocol">
```
Comment (thisisjoshford) [inline docs/develop/node/running-a-node.md:189]: The "Got a question" prompt with link to SO should be at the bottom of the document.

File: docs/develop/node/running-a-node.md [hunk 2]
```diff
@@ -188,3 +188,54 @@ To check logs call: `nearup logs` or `nearup logs --follow`
...
   <h8>Ask it on StackOverflow!</h8></a>
+
+## Starting a node from backup
+There are public backups of data available as tar archives.
+This enables you to sync data from the network faster.
+We have regular backups and archival backups.
+We also keep those for two chains, mainnet and testnet.
```
Comment (thisisjoshford) [inline docs/develop/node/running-a-node.md:196]:
```suggestion
Using data backups allows you to sync your node quickly by using public tar backup files. There are two types of backups for available for both `testnet` and `mainnet`:
- regular 
- archival 
```

Comment (thisisjoshford) [reply docs/develop/node/running-a-node.md:196]: Maybe we could add a description for regular and archival so people know what the difference between the two are.

File: docs/develop/node/running-a-node.md [hunk 3]
```diff
@@ -188,3 +188,54 @@ To check logs call: `nearup logs` or `nearup logs --follow`
...
+
+https://near-protocol-public.s3.ca-central-1.amazonaws.com/backups/mainnet/archive/data.tar
+
+Testnet regular data
+
+https://near-protocol-public.s3.ca-central-1.amazonaws.com/backups/testnet/rpc/data.tar
+
+Testnet archival data
+
+https://near-protocol-public.s3.ca-central-1.amazonaws.com/backups/testnet/archive/data.tar
```
Comment (thisisjoshford) [inline docs/develop/node/running-a-node.md:214]:
```suggestion
`mainnet` 
  - [regular](https://near-protocol-public.s3.ca-central-1.amazonaws.com/backups/mainnet/rpc/data.tar)
  - [archival](https://near-protocol-public.s3.ca-central-1.amazonaws.com/backups/mainnet/archive/data.tar)

`testnet`
  - [regular](https://near-protocol-public.s3.ca-central-1.amazonaws.com/backups/testnet/rpc/data.tar)
  - [archival](https://near-protocol-public.s3.ca-central-1.amazonaws.com/backups/testnet/archive/data.tar)
```

File: docs/develop/node/running-a-node.md [hunk 4]
```diff
@@ -188,3 +188,54 @@ To check logs call: `nearup logs` or `nearup logs --follow`
...
+
```
Comment (thisisjoshford) [inline docs/develop/node/running-a-node.md:215]:
```suggestion

### Starting the node
```

File: docs/develop/node/running-a-node.md [hunk 5]
```diff
@@ -188,3 +188,54 @@ To check logs call: `nearup logs` or `nearup logs --follow`
...
+Starting node from backup data neard
```
Comment (thisisjoshford) [inline docs/develop/node/running-a-node.md:216]:
```suggestion
Starting node using `neard` backup data 

```

File: docs/develop/node/running-a-node.md [hunk 6]
```diff
@@ -188,3 +188,54 @@ To check logs call: `nearup logs` or `nearup logs --follow`
...
+```bash
+./neard init --chain-id <chain-id> --download-genesis
+cd ~/.near/data
+wget <link-above>
+tar -xf data.tar
+rm data.tar
+./neard run
+```
+
+Starting node from backup data nearup
```
Comment (thisisjoshford) [inline docs/develop/node/running-a-node.md:226]:
```suggestion
Starting node using `nearup` backup data:

```

File: docs/develop/node/running-a-node.md [hunk 7]
```diff
@@ -188,3 +188,54 @@ To check logs call: `nearup logs` or `nearup logs --follow`
...
+nearup run <chain-id> && sleep 30 && nearup stop
+cd ~/.near/<chain-id>/data
+rm ./* # clean up old DB files to avoid corruption
+wget <link-above>
+tar -xf data.tar
+rm data.tar
+nearup run <chain-id> 
+```
+
+<chain-id> corresponds to testnet or mainnet
```
Comment (thisisjoshford) [inline docs/develop/node/running-a-node.md:237]:
```suggestion
( <chain-id> corresponds to `testnet` or `mainnet` )
```

File: docs/develop/node/running-a-node.md [hunk 8]
```diff
@@ -188,3 +188,54 @@ To check logs call: `nearup logs` or `nearup logs --follow`
...
+
+by default neard stores data in ~/.near/data
+
+nearup stores data in ~/.near/<chain-id>/data
```
Comment (thisisjoshford) [inline docs/develop/node/running-a-node.md:241]:
```suggestion
**Note:** Default location for `neard` data is `~/.near/data`. `nearup` stores data by default in `~/.near/<chain-id>/data.`
```

**Review by thisisjoshford — CHANGES_REQUESTED** [review]

File: docs/develop/node/running-a-node.md [hunk 9]
```diff
@@ -146,9 +146,9 @@ Enter your account ID (leave empty if not going to be a validator):
 
 Create new instance, with at least:
 
-* 2 vCPU and 3.75 GB of RAM (We recommend n1-standard-2).
+* 8 vCPU and 32 GB of RAM (We recommend e2-standard-8).
```
[the remark this answers is not in the document]
Comment (frol) [reply docs/develop/node/running-a-node.md:149]: Why do we need 8 vCPU? 32GB of RAM should not be a requirement as well. Can we be more constraint about our recommendations?

[the remark this answers is not in the document]
Comment (bowenwang1996) [reply docs/develop/node/running-a-node.md:149]: 4cpu and 8GB of ram should be enough. Even 2 cpus could be fine

[the remark this answers is not in the document]
Comment (mhalambek) [reply docs/develop/node/running-a-node.md:149]: Just took the ones we're using for official rpc nodes, can align them with @bowenwang1996 recommendations

File: docs/develop/node/running-a-node.md (hunk 4 above)
Comment (mhalambek) [reply docs/develop/node/running-a-node.md]: Solved most of the changes, lost context for this one, could you review this PR again

File: docs/develop/node/running-a-node.md [hunk 10]
```diff
@@ -185,6 +185,49 @@ Node is running!
...
+wget <link-above>
+tar -xf data.tar
+rm data.tar
+./neard run
+```
+
+Starting node using `nearup` backup data:
+
+```bash
+nearup run <chain-id> && sleep 30 && nearup stop
```
Comment (frol) [inline docs/develop/node/running-a-node.md:217]: `sleep 30` won't be enough to download testnet genesis records, won't this break things?

**Review by frol — APPROVED** [review]

**Comment by frol** [comment]
@mhalambek Can you also update https://docs.near.org/docs/roles/integrator/exchange-integration#running-an-archival-node ? (I suggest we merge the information into the page you already updated and just add a direct link from the Exchange Integration page)

**Comment by janewang** [comment]
@mhalambek This is a wonderful PR for starting a node from backup. Would be great to have this in the node documentation.

I merged a node documentation PR today and I think it may cause a number of merge conflicts. Please let me know if you'd like me to work on this PR and I can put your bash scripts into the docs. Thanks!

**Comment by janewang** [comment]
> @mhalambek Can you also update https://docs.near.org/docs/roles/integrator/exchange-integration#running-an-archival-node ? (I suggest we merge the information into the page you already updated and just add a direct link from the Exchange Integration page)

@frol This is a good suggestion. I'm going to open a PR to move the config for archival node to the consolidated node documentation.

File: docs/develop/node/running-a-node.md (hunk 4 above)
Comment (thisisjoshford) [reply docs/develop/node/running-a-node.md]: lol... not sure what this is either... disregard :)

**Pull request merged**